realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -138,36 +159,57 @@ void RealSenseNodeFactory::onInit()
 			cfg.enable_all_streams();
 			pipe->start(cfg); //File will be opened in read mode at this point
 			_device = pipe->get_active_profile().get_device();
-			_realSenseNode = std::unique_ptr<BaseRealSenseNode>(new BaseRealSenseNode(nh, privateNh, _device, _serial_no));
+			_realSenseNode = std::shared_ptr<BaseRealSenseNode>(new BaseRealSenseNode(nh, privateNh, _device, _serial_no));
 			_realSenseNode->publishTopics();
 			_realSenseNode->registerDynamicReconfigCb(nh);
 		}
 		else
 		{
 			privateNh.param("initial_reset", _initial_reset, false);
-			_device = getDevice();
-			if (_device && _initial_reset)
+
+			_is_alive = true;
+			_query_thread = std::thread([=]()
+						{
+							ROS_DEBUG("Waiting for device...");
+							std::chrono::milliseconds timespan(6000);
```

Comment:
Magic number `6000`? How was the value derived, empirically?

Since you have already have introduced a new *rosparm*, I recommend also changing this into a configurable parameter

Reply:
I copied this section of code from the latest intel version: https://github.com/IntelRealSense/realsense-ros/blob/development/realsense2_camera/src/realsense_node_factory.cpp#L272

I'm fine with adding a parameter though.

Reply:
Alright, assuming upstream changes are designed based on experiments and validated